src/main/scala/io/citrine/lolo/trees/splits/BoltzmannSplitter.scala

```diff
@@ -0,0 +1,226 @@
+package io.citrine.lolo.trees.splits
+
+import io.citrine.lolo.trees.impurity.VarianceCalculator
+import io.citrine.lolo.trees.splits.BoltzmannSplitter.SplitterResult
+
+import scala.collection.mutable
+import scala.util.Random
+
+/**
+  * Find a split for a regression problem
+  *
+  * The splits are picked with a probability that is related to the reduction in variance:
+  * P(split) ~ exp[ - {reduction in variance} / ({temperature} * {total variance}) ]
+  * recalling that the "variance" here is weighted by the sample size (so its really the sum of the square difference
+  * from the mean).  This is akin to kinetic monte carlo and simulated annealing techniques.
```

Comment:
I'm confused as to exactly which values are weighted and which are normalized.
Can I interpret `temperature` consistently independent of scale and sample size? My default interpretation is "if `temperature = 0.02`, then any split that reduces the relative variance by about 2% or more is equally likely to be picked." Is that correct? And if I train a model with a given temperature and then train a different model where I've scaled all the labels by a factor of 10 or used 10 times as much training data (drawn from the same underlying distribution), can I use the same temperature and expect the same results?

Reply:
Yeah; this language is simply wrong.  I'm revising for correctness and clarity.